docs/content/NewButton.mdx

```diff
@@ -0,0 +1,169 @@
+---
+title: Buttons
+status: Alpha
+source: https://github.com/primer/react/tree/main/src/NewButton
+storybook: '/react/storybook?path=/story/composite-components-new-button--default-button'
+description: Button is core design component used for actions.
+---
+
+import {Props} from '../src/props'
+import {ComponentChecklist} from '../src/component-checklist'
+import {NewButton} from '@primer/components'
+
+## Usage
+
+### Installation
+
+```js
+import {NewButton} from '@primer/components/drafts'
+```
+
+### Default button
+
+Using the component without any explicit props gives you the default button.
+
+```jsx live
+<NewButton>Default</NewButton>
+```
+
+### Danger button
+
+This button is used to warn users about potentially destructive actions
+
+```jsx live
+<NewButton variant="danger">Danger</NewButton>
+```
+
+### Outline button
+
+This button is typically used as a secondary button
+
+```jsx live
+<NewButton variant="outline">Outline</NewButton>
+```
+
+### Invisible button
+
+This button indicates that the action is a low priority one.
+
+```jsx live
+<NewButton variant="invisible">Invisible</NewButton>
+```
+
+### Different sized buttons
+
+Buttons can be in three different sizes. `small`, `medium`, `large`.
+
+```jsx live
+<>
+  <NewButton size="small">Search</NewButton>
+  <NewButton sx={{mt: 2}}>Search</NewButton>
+  <NewButton sx={{mt: 2}} size="large">
+    Search
+  </NewButton>
+</>
+```
+
+### Icon and button
+
+We can place an icon inside the button in either the leading or the trailing position
+
+```jsx live
+<>
+  <NewButton leadingIcon={SearchIcon}>Search</NewButton>
+  <NewButton trailingIcon={SearchIcon} sx={{mt: 2}}>
+    Search
+  </NewButton>
+</>
+```
+
+### Icon only button
+
+This uses the `icon` prop and creates a nice square button with only the icon visible.
+
+```jsx live
+<NewButton icon={SearchIcon}>Search</NewButton>
+```
+
+### Different sized icon buttons
+
+Buttons can be in three different sizes. `small`, `medium`, `large`.
+
+```jsx live
+<>
+  <NewButton icon={SearchIcon} size="small">
+    Search
+  </NewButton>
+  <NewButton icon={SearchIcon} sx={{mt: 2}}>
+    Search
+  </NewButton>
+  <NewButton icon={SearchIcon} sx={{mt: 2}} size="large">
+    Search
+  </NewButton>
+</>
+```
+
+### Counter component
+
+A common use case for primer is a button with a counter component which shows count value as provided by the consumer.
+We provide `Button.Counter` as a composite component which requires you to provide a number as child.
+It renders a counter with the same `variant` styles as the parent Button component
+
+```jsx live
+<NewButton>
+  Watch
+  <NewButton.Counter>1</NewButton.Counter>
```

Comment:
Love to see it!

Reply:
What happens if people render multiple `Button.Counter`s?

Reply:
Yeah.. The button api can be broken in multiple ways right now. You can basically provide any child component, or multiple `Button.Counter`. There's no design reason for it to have multiple counters. Don't think we need to account for those kind of usages.